content/rust/concepts/data-types/data-types.md

```diff
@@ -0,0 +1,55 @@
+---
+Title: 'Data Types in Rust'
+Description: 'An overview of data types in Rust, including scalar and compound types.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Rust'
+  - 'Data Types'
+CatalogContent:
+  - 'rust-for-programmers'
+  - 'paths/computer-science'
+---
+
+In Rust, data types classify the types of values that can be used and manipulated within a program. Rust's type system ensures that data types are known at compile time, which helps catch errors early.
+
+## Scalar Types
+
+Rust has four primary scalar types:
+- **Integers**: Represent whole numbers. Examples include `i32`, `u32`, `i64`, etc.
+- **Floating-Point Numbers**: Represent numbers with fractional components. Examples include `f32`, `f64`.
+- **Booleans**: Represent true or false values, using the `bool` type.
+- **Characters**: Represent single Unicode characters, using the `char` type.
```

Comment:
Can we elaborate on these points a little bit? Example:

- **Integers**: Represent whole numbers. The signed integer starts with an `i` followed by the bit number and the unsigned integer starts with `u` followed by the bit number. The built-in integer types in Rust include `8bit`, `16bit`, `32bit`, `64bit`, and `128bit`.

You can elaborate a bit on the other bullet points as well. (Not much, just the representation and what you think could be useful to a beginner reading this.)